src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -324,7 +324,16 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                             genHWIntrinsic_R_R_RM_R(node, ins);
                             break;
                         }
+                        case NI_AVXVNNI_MultiplyWideningAndAdd:
+                        case NI_AVXVNNI_MultiplyWideningAndAddSaturate:
+                        {
+                            assert(targetReg != REG_NA);
```

Comment:
These assertions are not really needed here -  we would check the same invariants at the beginning of `genHWIntrinsic_R_R_R_RM`